aws_lambda_builders/workflows/python_pip/validator.py

```diff
@@ -48,11 +50,24 @@ def validate(self, runtime_path):
             cmd, cwd=os.getcwd(), stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=OSUtils().original_environ()
         )
         p.communicate()
+
         if p.returncode != 0:
             raise MisMatchRuntimeError(language=self.language, required_runtime=self.runtime, runtime_path=runtime_path)
-        else:
-            self._valid_runtime_path = runtime_path
-            return self._valid_runtime_path
+
+        try:
+            # call method to import `pip`
+            # ignoring method return values since we only want to check
+            # if `pip` is imported successfully
+            pip_import_string(runtime_path)
```

Comment:
I think it is better if we extract the part where we execute the python command to import pip in a separate function, and to have a function to check if pip is installed or not, but any way it is not a blocking

Reply:
Based on the below feedback (the "what-if" case), this will be moved out of this validator and into the build action so that this only finds the pip runtime if things are being built.